integration_test/integration_test.go

```diff
@@ -290,13 +298,199 @@ func (t *WASMIntegrationTestSuite) Test04_ContractExecution() {
 }
 
+func (i *EVMIntegrationTestSuite) TeardownTest() {}
+
+func (t *EVMIntegrationTestSuite) Test01_CheckBalance() {
+	resp, err := t.EthClient.BalanceAt(context.Background(), t.UserWallet1.EthAddress, nil)
+
+	assert.NoError(t.T(), err)
+
+	expectedInt := new(big.Int)
+	expectedInt, _ = expectedInt.SetString("99000000000000000000", 10)
+	assert.GreaterOrEqual(t.T(), 1, resp.Cmp(expectedInt))
+
+	expectedInt, _ = expectedInt.SetString("100000000000000000000", 10)
+	assert.LessOrEqual(t.T(), -1, resp.Cmp(expectedInt))
+}
+
+func (t *EVMIntegrationTestSuite) Test02_DeployTokenContract() {
+	// Prepare parameters
+	networkId, err := t.EthClient.NetworkID(context.Background())
+	assert.NoError(t.T(), err)
+
+	ethPrivkey, _ := ethcrypto.ToECDSA(t.UserWallet1.CosmosWalletInfo.PrivKey.Bytes())
+	auth, err := abibind.NewKeyedTransactorWithChainID(ethPrivkey, networkId)
+	assert.NoError(t.T(), err)
+
+	auth.GasLimit = uint64(1300000)
+	auth.GasPrice, _ = new(big.Int).SetString(xplaGasPrice, 10)
+
+	strbin, err := os.ReadFile(filepath.Join(".", "misc", "token.sol.bin"))
+	assert.NoError(t.T(), err)
+
+	binbyte, _ := hex.DecodeString(string(strbin))
+
+	parsedAbi, err := TokenInterfaceMetaData.GetAbi()
+	assert.NoError(t.T(), err)
+	assert.NotNil(t.T(), parsedAbi)
+
+	// Actual deploy
+	address, tx, _, err := abibind.DeployContract(auth, *parsedAbi, binbyte, t.EthClient, "Example Token", "XPLAERC")
+	assert.NoError(t.T(), err)
+	fmt.Println("Tx hash: ", tx.Hash().String())
+
+	time.Sleep(time.Second * 7)
+
+	fmt.Println("Token address: ", address.String())
+	t.TokenAddress = address
+}
+
+func (t *EVMIntegrationTestSuite) Test03_ExecuteTokenContractAndQueryOnEvmJsonRpc() {
+	// Prepare parameters
+	networkId, err := t.EthClient.NetworkID(context.Background())
+	assert.NoError(t.T(), err)
+
+	store, err := NewTokenInterface(t.TokenAddress, t.EthClient)
+	assert.NoError(t.T(), err)
+
+	// 10^18
+	multiplier, _ := new(big.Int).SetString("1000000000000000000", 10)
+	amt := new(big.Int).Mul(big.NewInt(10), multiplier)
+
+	ethPrivkey, _ := ethcrypto.ToECDSA(t.UserWallet1.CosmosWalletInfo.PrivKey.Bytes())
+	auth, err := abibind.NewKeyedTransactorWithChainID(ethPrivkey, networkId)
+	assert.NoError(t.T(), err)
+
+	auth.GasLimit = uint64(300000)
+	auth.GasPrice, _ = new(big.Int).SetString(xplaGasPrice, 10)
+
+	// try to transfer
+	tx, err := store.Transfer(auth, t.UserWallet2.EthAddress, amt)
+	assert.NoError(t.T(), err)
+	fmt.Println("Sent as ", tx.Hash().String())
+
+	time.Sleep(time.Second * 7)
+
+	// query & assert
+	callOpt := &abibind.CallOpts{}
+	resp, err := store.BalanceOf(callOpt, t.UserWallet2.EthAddress)
+	assert.NoError(t.T(), err)
+
+	assert.Equal(t.T(), amt, resp)
+}
+
+// Wrote and tried to test triggering EVM by MsgEthereumTx
+// But there is a collision between tx msg caching <> ethermint antehandler
+// MsgEthereumTx.From kept left <> ethermint antehandler checks and passes only MsgEthereumTx.From is empty
+// It resolves from ethermint v0.20
```

Comment:
Ethermint issue: https://github.com/evmos/ethermint/pull/1479

> It can be executed only with the tx sign of cosmos without the sign in the msg?
I don't know because I made the msg from the signed ETH tx